src/api/services/userService.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { NextFunction, Response } from 'express';
+import { Knex } from 'knex';
+
+import { TestConfigure } from '../../database/TestSelfServeDatabase';
+import {
+  createParticipant,
+  createResponseObject,
+  createUser,
+  createUserParticipantRequest,
+  getParticipantIdsOfUser,
+} from '../../testHelpers/apiTestHelpers';
+import { verifyAndEnrichUser } from '../middleware/usersMiddleware';
+import * as kcUsersService from './kcUsersService';
+import { UserService } from './userService';
+
+describe('User Service Tests', () => {
+  let knex: Knex;
+  let next: NextFunction;
+  let res: Response;
+
+  beforeEach(async () => {
+    knex = await TestConfigure();
+    next = jest.fn();
+    ({ res } = createResponseObject());
+    jest.spyOn(kcUsersService, 'removeApiParticipantMemberRole').mockResolvedValue();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('User removal', () => {
+    describe('User still belongs to a participant', () => {
+      it('removes the user from the correct participant', async () => {
+        const targetParticipant = await createParticipant(knex, {});
+        const anotherParticipant = await createParticipant(knex, {});
+        const user = await createUser({
+          participantToRoles: [
+            { participantId: targetParticipant.id },
+            { participantId: anotherParticipant.id },
+          ],
+        });
+        const request = createUserParticipantRequest(user.email, targetParticipant, user.id);
+        await verifyAndEnrichUser(request, res, next);
+
+        const userService = new UserService();
+        await userService.removeUser(request);
+
+        const userParticipantIds = await getParticipantIdsOfUser(user.email);
+        expect(userParticipantIds).not.toContain(targetParticipant.id);
+        expect(userParticipantIds).toContain(anotherParticipant.id);
+      });
+    });
+    it('user keycloak roles are unchanged', async () => {
+      const targetParticipant = await createParticipant(knex, {});
```

Comment:
Can you reuse come code between these two tests, for example, defining `targetParticipant` , `anotherParticipant` and `user` outside of the it block?

Reply:
Or can you even just combine it all into one test and assert multiple things?  Seems inefficient to do the exact same setup in multiple tests.

Reply:
I did try factor out some shared setup but functions directly in the `describe` block cannot be awaited because jest [prevents async describes](https://github.com/jest-community/eslint-plugin-jest/blob/v25.7.0/docs/rules/valid-describe-callback.md). 

I've gone with Scott's suggestion to just combine them into fewer tests. My original intention was to keep the assertions separate but I think the tradeoff for readability is fine.